homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -1,27 +1,24 @@
 """Support for Fritzbox binary sensors."""
-import logging
-
 import requests
 
-from homeassistant.components.binary_sensor import BinarySensorDevice
-
-from . import DOMAIN as FRITZBOX_DOMAIN
+from homeassistant.components.binary_sensor import DOMAIN, BinarySensorDevice
+from homeassistant.const import CONF_DEVICES
 
-_LOGGER = logging.getLogger(__name__)
+from .const import CONF_CONNECTIONS, DOMAIN as FRITZBOX_DOMAIN, LOGGER
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Fritzbox binary sensor platform."""
+async def async_setup_entry(hass, config_entry, async_add_entities):
```

Comment:
For **another PR** we could consider consolidating the platform setup, since it's very similar for all platforms. The only things that change are the device `.has_*` property check and the entity class.